autotx/helper_agents/user_proxy.py

```diff
@@ -1,13 +1,16 @@
 from textwrap import dedent
-from typing import Any, Callable, Dict, Optional
+from typing import TYPE_CHECKING, Any, Callable, Dict, Optional
 from autogen import UserProxyAgent
 
-def build(user_prompt: str, agents_information: str, get_llm_config: Callable[[], Optional[Dict[str, Any]]]) -> UserProxyAgent:
+if TYPE_CHECKING:
+    from autotx.AutoTx import CustomModel
+
+def build(user_prompt: str, agents_information: str, get_llm_config: Callable[[], Optional[Dict[str, Any]]], custom_model: Optional['CustomModel']) -> UserProxyAgent:
     user_proxy = UserProxyAgent(
         name="user_proxy",
         is_termination_msg=lambda x: x.get("content", "") and "TERMINATE" in x.get("content", ""),
         human_input_mode="NEVER",
-        max_consecutive_auto_reply=10,
+        max_consecutive_auto_reply=4 if custom_model else 10,
```

Comment:
Interesting, why this change?

Reply:
the user proxy is not able to consistently terminate when the goal has been done - in order to avoid a lot of time of execution, we put a lower number for auto replies